### PR TITLE
go_snappy build tag to toggle snappy library

### PR DESCRIPTION
**Pull request by abe-winter-4s**
* use `github.com/golang/snappy` instead of `github.com/cockroachdb/c-snappy` when run with `go build -tags native_snappy`
* standard invocation (i.e. `go build`) will be unchanged

### CODE REVIEW

**Comment by theevocater** [comment]
ugh i can fix the travis stuff.

**Comment by abe-winter-4s** [comment]
should we postpone fixing this until after perf tests?

**Comment by theevocater** [comment]
nah its a simple fix, one sec

**Comment by theevocater** [comment]
this _should_ work if you rebase on master

File: hfile/c_reader.go
Comment (theevocater) [inline hfile/c_reader.go]: my only nit is I'd call these c_snappy.go and go_snappy.go

File: hfile/go_reader.go
Comment (theevocater) [inline hfile/go_reader.go]: maybe add a quick comment as to why this happens? just in case we somehow stop working on this and forget forever

**Review by theevocater — CHANGES_REQUESTED** [review]

**Review by theevocater — APPROVED** [review]

**Comment by theevocater** [comment]
oh we should probably lock down the version of gosnappy at some point, if we decide to keep it, but otherwise LGTM 👍

**Comment by abe-winter-4s** [comment]
was holding off pending a decision about govendor / godep. Should I pin it before merging?

**Comment by theevocater** [comment]
nah its fine, we can pin it when we decide to start using it. until then its not even part of the build!

**Pull request merged**
